pkg/distributor/distributor.go

```diff
@@ -78,6 +78,12 @@ var (
 	})
 )
 
+type clusterTracker interface {
+	LookupReplica(ctx context.Context, cluster string, replica string) *replicaTracker
+	setTimeout(overwrite, write int64)
+	loop(ctx context.Context)
```

Comment:
Lets have the constructor do the `go loop()` so it doesn't have to be in the interface.  You'll want to add a `Stop` though.